File: src/main/java/org/springframework/samples/petclinic/rest/mapper/VisitMapper.java
package org.springframework.samples.petclinic.rest.mapper;

import org.mapstruct.Mapper;
import org.springframework.samples.petclinic.owner.Visit;
import org.springframework.samples.petclinic.rest.dto.VisitDto;

@Mapper(componentModel = "spring")
public interface VisitMapper {

	VisitDto toDto(Visit visit);

}

File: src/main/kotlin/org/springframework/samples/petclinic/rest/mapper/VisitMapper.kt
package org.springframework.samples.petclinic.rest.mapper

import org.springframework.samples.petclinic.owner.Visit
import org.springframework.samples.petclinic.rest.dto.VisitDto
import org.springframework.stereotype.Component

@Component
class VisitMapper {

    fun toDto(visit: Visit): VisitDto {
        return VisitDto(
            id = visit.id,
            date = visit.date,
            description = visit.description
        )
    }

}